tsl/test/t/002_chunk_copy_move.pl

```diff
@@ -28,33 +28,59 @@
     ]);
 
 #Check that chunks are shown appropriately on all nodes of the multi-node setup
-my $query = q[SELECT * from show_chunks('test');];
+my $query        = q[SELECT * from show_chunks('test');];
+my $operation_id = "ts_copy_1_1";
 
-#Query Access node
-$an->psql_is(
-	'postgres', $query, q[_timescaledb_internal._dist_hyper_1_1_chunk
-_timescaledb_internal._dist_hyper_1_2_chunk
-_timescaledb_internal._dist_hyper_1_3_chunk
-_timescaledb_internal._dist_hyper_1_4_chunk], 'AN shows correct set of chunks'
-);
+#Check chunk states before the move
+check_pre_move_chunk_states();
 
-#Query datanode1
-$dn1->psql_is(
+#Setup the error injection function on the AN
+my $extversion = $an->safe_psql('postgres',
+	"SELECT extversion from pg_catalog.pg_extension WHERE extname = 'timescaledb'"
+);
+$an->safe_psql(
 	'postgres',
-	$query,
-	"_timescaledb_internal._dist_hyper_1_1_chunk\n_timescaledb_internal._dist_hyper_1_3_chunk\n_timescaledb_internal._dist_hyper_1_4_chunk",
-	'DN1 shows correct set of chunks');
-
-#Check contents on the chunk on DN1
-$dn1->psql_is('postgres',
-	"SELECT sum(device) FROM _timescaledb_internal._dist_hyper_1_1_chunk",
-	qq[406], "DN1 has correct contents in the chunk");
+	qq[
+    CREATE OR REPLACE FUNCTION error_injection_on(TEXT) RETURNS VOID LANGUAGE C VOLATILE STRICT
+    AS 'timescaledb-$extversion', 'ts_debug_point_enable';
+    ]);
 
-#Query datanode2
-$dn2->psql_is(
-	'postgres', $query,
-	"_timescaledb_internal._dist_hyper_1_2_chunk",
-	'DN2 shows correct set of chunks');
+#Induce errors in various stages in the chunk move activity and ensure that the
+#cleanup function restores things to the previous sane state
+
+my @stages =
+  qw(init create_empty_chunk create_publication create_replication_slot create_subscription sync_start sync drop_publication drop_subscription attach_chunk delete_chunk);
+
+my ($stdout, $stderr, $ret);
+my $curr_index = 1;
+my $arrSize    = @stages;
+
+while ($curr_index < $arrSize)
+{
+	#Enable the error at each stage
+	#Call the move_chunk procedure which should error out now
+	($ret, $stdout, $stderr) = $an->psql('postgres',
+		"SELECT error_injection_on('$stages[$curr_index]'); CALL timescaledb_experimental.move_chunk(chunk=>'_timescaledb_internal._dist_hyper_1_1_chunk', source_node=> 'dn1', destination_node => 'dn2');"
+	);
+	is($ret, 3,
+		"move_chunk fails as expected in stage '$stages[$curr_index]'");
+	like(
+		$stderr,
+		qr/ERROR:  error injected at debug point '$stages[$curr_index]'/,
+		'failure in expected stage');
+
+	#The earlier debug error point gets released automatically since it's a session lock
+	#Call the cleanup procedure to make things right
+	$operation_id = "ts_copy_" . $curr_index . "_1";
+	$an->safe_psql('postgres',
+		"CALL timescaledb_experimental.copy_chunk_cleanup(operation_id=>'$operation_id');"
+	);
+
+	#Check chunk state is as before the move
+	check_pre_move_chunk_states();
```

Comment:
nit: just a thought that this could be tested in the same way as we run stages. Basically add a separate function which check the state after cleanup and iterate over them instead

Reply:
Didn't get this.

Reply:
I thought it makes sense to test each stage separately, but seems to be it will be complicated. I guess we need only to test that it properly transition to the correct stage after cleanup, which is there.